net-nds/389-ds-base/389-ds-base-3.0.2.ebuild

```diff
@@ -242,10 +242,10 @@ src_configure() {
 
 	# https://github.com/389ds/389-ds-base/issues/4292 part 2
 	# creates wrongly named unit file if == no
-	use systemd && myconf+=(
-		$(use_with systemd)
-		$(use_with systemdgroupname "dirsrv.target")
-		$(use_with tmpfiles-d "/usr/lib/tmpfiles.d")
+	use systemd && myeconfargs+=(
+		--with-systemd
```

Comment:
Some comments:
1. `--with-systemd` should be hoisted into the above array to avoid automagic libsystemd usage
2. I suspect that all of those configure arguments should actually be set unconditionally. The eclass handles systemd not being installed for `systemd_get_systemunitdir`.

Reply:
Moved everything into the main econf variable and looks like it handles it correctly. Tried it on a system with and one without systemd.

Command configure with USE flag systemd set:
```
./configure
--prefix=/usr
--build=x86_64-pc-linux-gnu
--host=x86_64-pc-linux-gnu
--mandir=/usr/share/man
--infodir=/usr/share/info
--datadir=/usr/share
--sysconfdir=/etc
--localstatedir=/var/lib
--datarootdir=/usr/share
--disable-dependency-tracking
--disable-silent-rules
--disable-static
--docdir=/usr/share/doc/389-ds-base-3.0.2
--htmldir=/usr/share/doc/389-ds-base-3.0.2/html
--with-sysroot=/
--libdir=/usr/lib64
--enable-acctpolicy
--enable-bitwise
--enable-dna
--enable-pam-passthru
--enable-autobind
--disable-auto-dn-suffix
--disable-debug
--enable-ldapi
--with-selinux
--without-initddir <--
--with-systemd <--
--disable-cmocka
--with-systemdgroupname=dirsrv.target <--
--with-tmpfiles-d=/lib/tmpfiles.d <--
--with-systemdsystemunitdir=/usr/lib/systemd/system <--
--enable-rust-offline
--with-pythonexec=/usr/bin/python3.12
--with-fhs
--with-openldap
--with-db-inc=/usr/include/db5.3
--disable-cockpit
```

Without USE flag systemd set:
```
./configure
--prefix=/usr
--build=x86_64-pc-linux-gnu
--host=x86_64-pc-linux-gnu
--mandir=/usr/share/man
--infodir=/usr/share/info
--datadir=/usr/share
--sysconfdir=/etc
--localstatedir=/var/lib
--datarootdir=/usr/share
--disable-dependency-tracking
--disable-silent-rules
--disable-static
--docdir=/usr/share/doc/389-ds-base-3.0.2
--htmldir=/usr/share/doc/389-ds-base-3.0.2/html
--with-sysroot=/
--libdir=/usr/lib64
--enable-acctpolicy
--enable-bitwise
--enable-dna
--enable-pam-passthru
--enable-autobind
--disable-auto-dn-suffix
--disable-debug
--enable-ldapi
--without-selinux
--with-initddir=/etc/init.d <--
--without-systemd <--
--disable-cmocka
--with-systemdgroupname=dirsrv.target <--
--with-tmpfiles-d=/lib/tmpfiles.d <--
--with-systemdsystemunitdir=/lib/systemd/system <--
--enable-rust-offline
--with-pythonexec=/usr/bin/python3.12
--with-fhs
--with-openldap
--with-db-inc=/usr/include/db5.3
--disable-cockpit
```

It does install systemd related files even if `--without-systemd` is set.
```
# emerge -pv 389-ds-base
[ebuild   R   ~] net-nds/389-ds-base-3.0.2::anders-larsson  USE="accountpolicy autobind bitwise dna ldapi pam-passthru -auto-dn-suffix -debug -doc (-selinux) -systemd -test" PYTHON_SINGLE_TARGET="python3_12 -python3_10 -python3_11" 0 KiB
# equery f 389-ds-base | grep systemd
/lib/systemd
/lib/systemd/system
/lib/systemd/system/dirsrv-snmp.service
/lib/systemd/system/dirsrv.target
/lib/systemd/system/dirsrv@.service
/lib/systemd/system/dirsrv@.service.d
/lib/systemd/system/dirsrv@.service.d/custom.conf
/usr/libexec/ds_systemd_ask_password_acl
/usr/share/man/man5/dirsrv.systemd.5.bz2
```

However, maybe this is not a problem. Hadn't reflected on it but there are quite a few systemd services installed into /lib/systemd/system on this openrc system.

Changes used:
```
--- a/net-nds/389-ds-base/389-ds-base-3.0.2.ebuild
+++ b/net-nds/389-ds-base/389-ds-base-3.0.2.ebuild
@@ -231,7 +231,11 @@ src_configure() {
                $(use_enable ldapi)
                $(use_with selinux)
                $(use_with !systemd initddir "/etc/init.d")
+               $(use_with systemd)
                $(use_enable test cmocka)
+               --with-systemdgroupname="dirsrv.target"
+               --with-tmpfiles-d="${EPREFIX}/lib/tmpfiles.d"
+               --with-systemdsystemunitdir="$(systemd_get_systemunitdir)"
                --enable-rust-offline
                --with-pythonexec="${PYTHON}"
                --with-fhs
@@ -240,15 +244,6 @@ src_configure() {
                --disable-cockpit
        )

-       # https://github.com/389ds/389-ds-base/issues/4292 part 2
-       # creates wrongly named unit file if == no
-       use systemd && myconf+=(
-               $(use_with systemd)
-               $(use_with systemdgroupname "dirsrv.target")
-               $(use_with tmpfiles-d "/usr/lib/tmpfiles.d")
-               --with-systemdsystemunitdir="$(systemd_get_systemunitdir)"
-       )
-
        econf "${myeconfargs[@]}"

        rm .cargo/config || die
```

Reply:
Yeah, it's the 'small files policy'. See https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301.